packages/clerk-js/src/ui/elements/Menu.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { PropsWithChildren, useEffect, useRef } from 'react';
+
+import { createContextAndHook } from '../../ui/utils';
+import { Button, Col } from '../customizables';
+import { usePopover } from '../hooks';
+import { animations, PropsOfComponent } from '../styledSystem';
+import { colors } from '../utils/colors';
+import { Portal } from './Portal';
+
+type UsePopoverReturn = ReturnType<typeof usePopover>;
+
+type MenuState = {
+  popoverCtx: UsePopoverReturn;
+};
+
+const [MenuStateCtx, useMenuState] = createContextAndHook<MenuState>('MenuState');
+
+type MenuProps = PropsWithChildren<Record<never, never>>;
+
+export const Menu = (props: MenuProps) => {
+  const popoverCtx = usePopover({
+    placement: 'right-start',
+    offset: 8,
+  });
+
+  return (
+    <MenuStateCtx.Provider
+      value={{ value: { popoverCtx } }}
+      {...props}
+    />
+  );
+};
+
+type MenuButtonProps = PropsOfComponent<typeof Button>;
+
+export const MenuButton = (props: MenuButtonProps) => {
+  const { popoverCtx } = useMenuState();
+  const { reference, toggle } = popoverCtx;
+
+  return (
+    <Button
+      ref={reference}
+      onClick={toggle}
+      {...props}
+    />
+  );
+};
+
+const findMenuItem = (el: Element, siblingType: 'prev' | 'next', options?: { countSelf?: boolean }) => {
+  let tagName = options?.countSelf ? el.tagName : '';
+  let sibling: Element | null = el;
+  while (sibling && tagName !== 'BUTTON') {
+    sibling = sibling[siblingType === 'prev' ? 'previousElementSibling' : 'nextElementSibling'];
+    tagName = sibling?.tagName ?? '';
+  }
+  return sibling;
+};
+
+type MenuListProps = PropsOfComponent<typeof Col>;
+
+export const MenuList = (props: MenuListProps) => {
+  const { sx, ...rest } = props;
+  const { popoverCtx } = useMenuState();
+  const { floating, styles, isOpen } = popoverCtx;
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const current = containerRef.current;
+    floating(current);
```

Comment:
I'd expect `floating` to be passed to the `ref` prop but I guess this is a way to use `floating` while still having a custom reference (`containerRef`) you can use for `.focus`. Can you confirm that this is indeed the case?

Reply:
Yes, that's the case.